labgrid/driver/power/ubus.py

```diff
@@ -0,0 +1,82 @@
+""" UBUS jsonrpc interface for PoE management on OpenWrt devices. This comes in
+    handy if devices are connected to a PoE switch running OpenWrt. Author: Paul
+    Spooren <mail@aparcar.org>
+
+    The URL given in hosts in exporter.yaml must accept unauthenticated UBUS
+    calls for the two `poe` calls `info` and `manage`.
+
+    An ACL example is given below:
+
+    root@switch:~# cat /usr/share/rpcd/acl.d/unauthenticated.json 
+    {
+        "unauthenticated": {
+            "description": "Access controls for unauthenticated requests",
+            "read": {
+                "ubus": {
+                    "session": [
+                        "access",
+                        "login"
+                    ],
+                    "poe": [
+                        "info",
+                        "manage"
+                    ]
+                }
+            }
+        }
+    }
+
+    Further information is availbe at https://openwrt.org/docs/techref/ubus#acls
+
+  NetworkPowerPort:
+      model: ubus
+      host: 'http://192.168.1.1/ubus'
+      index: 1
+"""
+
+import requests
+
+
+def power_set(host, port, index, value):
+    assert port is None
+
+    r = requests.post(
```

Comment:
You could have a function which handles requests and pass it the command ('manage' or 'info') along with the extra dict. That would reduce code duplication.

Reply:
Done, thanks.